R/utils.R

```diff
@@ -52,6 +52,9 @@ save_to_synapse <- function(input_file,
                             parent,
                             name = NULL,
                             annotations = NULL) {
+  if (!is_name_valid(name)) {
+    stop("Please check that file names only contains: letters, numbers, spaces, underscores, hyphens, periods, plus signs, and parentheses.") # nolint
+  }
   file_to_upload <- synapser::File(
     input_file$datapath,
     parent = parent,
```

Comment:
I think we should change this function signature a little bit. It doesn't seem right to have a default value that we then say is invalid (even though in practice we usually override the default value). And the reason `name` defaults to `NULL` in `save_to_synapse()` was because I thought that was the default value of `File()`, but it turns out when you explicitly pass `NULL` to `File()` you change the behavior: https://sagebionetworks.jira.com/browse/SYNR-1437.

What do you think about removing `name` as an argument altogether, since we seem to always want to get it from the input file anyway?

```r
save_to_synapse <- function(input_file, parent, annotations = NULL) {
  file_to_upload <- synapser::File(
    input_file$datapath,
    parent = parent,
    name = input_file$name,
    annotations = annotations
  )
  if (!is_name_valid(file_to_upload$properties$name)) {
    stop("Please check that file names only contains: letters, numbers, spaces, underscores, hyphens, periods, plus signs, and parentheses.") # nolint
  }
  synapser::synStore(file_to_upload)
}
```

Reply:
That makes sense. My only question is that that changing the `save_to_synapse()` signature is a different issue and involves more changes that simply checking that the file name is valid. What do you think about me fixing the call to `is_name_valid()` to use the name from the `input_file` and then opening a separate issue to fix `save_to_synapse()`?

Reply:
Sounds good to me, thanks!

Reply:
Commit 8cca7e9 changes the call to `is_name_valid()` in `save_to_synapse()` to use `input_file$name` in preparation for changing the signature for `save_to_synapse()`. 

I left the test for a `NULL` name in both the `is_name_valid()` function and test-check-name.R. Did you want me to remove these checks? With how we are using this function, there should not be a case where `NULL` is sent to `is_name_valid()`, but there could be an edge case that I am missing.